src/interchange/avro.rs

```diff
@@ -612,13 +613,27 @@ impl Encoder {
                         .0;
                     let avro_val = Self::data_to_avro(node.step(&before.schema), &row.unpack())?;
                     // Add wrapper Record with before and after RecordFields
-                    let wrapped_avro_val = Value::Record(vec![
-                        (
-                            "before".into(),
-                            Value::Union(null_idx, Box::from(Value::Null)),
-                        ),
-                        ("after".into(), avro_val),
-                    ]);
+                    let wrapped_avro_val = if diff == -1 {
+                        Value::Record(vec![
+                            ("before".into(), avro_val),
+                            (
+                                "after".into(),
+                                Value::Union(null_idx, Box::from(Value::Null)),
+                            ),
+                        ])
+                    } else {
+                        if diff != 1 {
+                            error!("Received an invalid diff {} when encoding a avro record. Defaulting to encode as insert", diff);
```

Comment:
This error should be returned, not logged! We have no choice but to log in the dataflow package, but we should push the decision to log the error as high in the stack as possible. `interchange` is meant to be a well-behaved library crate, so it should just return the error. 

I'll send a follow-up PR.

Reply:
Yes, but the reason I'm logging here is because it wasn't clear to me or Frank what to do with diffs that have multiset semantics yet (i.e. converting diff = -4 to a single retraction is not quite right, but its also unclear if downstream consumers can take in 4 consecutive retractions or if thats even meaningful) and it seemed like a conservative choice to default to our old behaviour, emit an error, and fix in a different pr

Reply:
What do you think about possibly erroring if you try and create a debezium sink from a `RelationExpr` that doesn't have a key? That might be the right first moment to complain, back when someone is trying to express the query.

I've only just arrived here though, so if you have a few better ideas, excellent and don't mind me! :D

Reply:
#2465 will make it so we just count out the diffs in unary, which seems reasonable enough to me.

Reply:
Chime in there if you disagree though!